server/src/main/java/org/opensearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -1529,6 +1530,9 @@ private void validate(String name, @Nullable Settings settings, List<String> ind
                 Optional.empty()
             );
             validationErrors.addAll(indexSettingsValidation);
+
+            // validate index refresh interval settings
+            validateRefreshIntervalSettings(settings, clusterService.getClusterSettings());
```

Comment:
can you check   snapshot restore path as well  ?

Reply:
Thanks for catching this @gbbafna. I've updated the PR to handle snapshot restore flow for index and indextemplate